cli/remotebazel/remotebazel.go

```diff
@@ -78,6 +78,7 @@ var (
 	remoteRunner            = RemoteFlagset.String("remote_runner", defaultRemoteExecutionURL, "The Buildbuddy grpc target the remote runner should run on.")
 	timeout                 = RemoteFlagset.Duration("timeout", 0, "If set, requests that have exceeded this timeout will be canceled automatically. (Ex. --timeout=15m; --timeout=2h)")
 	execPropsFlag           = bbflag.New(RemoteFlagset, "runner_exec_properties", []string{}, "Exec properties that will apply to the *ci runner execution*. Key-value pairs should be separated by '=' (Ex. --runner_exec_properties=NAME=VALUE). Can be specified more than once. NOTE: If you want to apply an exec property to the bazel command that's run on the runner, just pass at the end of the command (Ex. bb remote build //... --remote_default_exec_properties=OSFamily=linux).")
+	remoteHeaders           = bbflag.New(RemoteFlagset, "remote_header", []string{}, "Remote headers to be applied to the execution request for the remote run. Can be used to set platform properties containing secrets (Ex. --remote_header=x-buildbuddy-platform.SECRET_NAME=SECRET_VALUE). Can be specified more than once.")
```

Comment:
for the flag name specifically, we might want to call it `--runner_remote_header`. The "runner" prefix helps clarify which flags are applied to the runner, and which flags are applied to the inner bazel invocation

For the APIs, it seems fine to omit the "runner" prefix since it seems like that's the convention we're currently using.